### ISSUE TITLE
Further development of nix-flake.el

### ISSUE THREAD

**Issue by akirak** [NixOS/nix-mode#148]
I would like to discuss `nix-flake.el` which I added to this repository in #140. Thank you for accepting the PR.

After using it for a while myself, I noticed that there is much room for improvement of it. One thing is asynchronously getting metadata of a flake for better experience, which is sometimes seriously necessary, but there are more.

Now I have some ideas for a comprehensive support of nix flakes, but it can become a big project and take too long. Maybe I will rewrite all of `nix-flake.el` in the future. I am not sure if I should include it in this repository.

Disclosure: I am working on [my personal project](https://github.com/akirak/emacs-twist). There is [a companion package](https://github.com/akirak/twist.el), and I also plan on developing a flake browser for it, which should become a separate package independent from the framework.

While it may be possible to make the flake browser as part of `nix-mode`, I think it is going too far. It will become more complex than the rest of nix-mode, which is probably too much of trouble. I think it is safer to start it as a separate package, so I will maintain it on my own.

My question is about `nix-flake.el` currently residing in this repository. Because I plan on rewriting it, it may be better to maintain it in a separate repository, owned by me. I will maintain it no matter where it is, but do you want to keep it in this repository, or should I make a separate repository for it? 

Thank you for maintaining nix-mode, and sorry for this issue.

**Comment by matthewbauer** [comment]
Separating out `nix-flake.el` would be fine by me. I don't think I have a strong preference either way but it's definitely the case that Emacs Lisp can require a lot of rapid development, so it's understandable to want it in your own repo. I don't actually have admin permissions on this repo, so I can't give you merge permissions but I think I can ask someone else in the NixOS org for it. I've based a lot of `nix-mode` around how `haskell-mode` is setup (https://github.com/haskell/haskell-mode), which tends to have a lot of experimental modes. I think if you'd like to make a nix-flake repo, we could leave the current version of nix-flake.el as is, but then have a pointer to use the newer one in case anyone is relying on `nix-mode` providing it.

BTW, twist looks very cool!

**Comment by akirak** [comment]
Thank you for your comment. I have created [a repo](https://github.com/emacs-twist/nix26.el) for the package. I agree with you that an own repo will be better for a faster feedback cycle. Just in case, I gave it the same license as nix-mode to allow merging in the future.

From now on, I will primarily work on `nix26.el`, but if there is a bug report on `nix-flake.el` in this repo, I will fix issues as far as I can.

> BTW, twist looks very cool!

Thanks, it freezes packages, so it must be cool.

And sorry for my English. I need better disciplines, which is one of the reasons I started using Emacs. Now I want a better integration with external programs, and Nix will be a key.
